salt/modules/aptpkg.py

```diff
@@ -172,7 +172,14 @@ def _call_apt(args, scope=True, **kwargs):
               'env': salt.utils.environment.get_module_environment(globals())}
     params.update(kwargs)
 
-    return __salt__['cmd.run_all'](cmd, **params)
+    cmd_ret = __salt__['cmd.run_all'](cmd, **params)
+    count = 0
+    while 'stderr' in cmd_ret and "Could not get lock" in cmd_ret['stderr'] and count < 10:
```

Comment:
This used to be a python anti-pattern, to do:
`if 'foo' in dict and 'blah' == dict['foo']`
the recommended would be:
`errText = cmd_ret.get('stderr')`
`if 'Could not get lock' in errText`

or directly:
`while 'Could not get lock' in cmd_ret.get('stderr', '')`

Reply:
yes it's better, I changed the condition by your example